update readme (#100)
vanniktech/gradle-maven-publish-plugin: README.md

Information: [This plugin is also available on Gradle plugins](https://plugins.gradle.org/plugin/com.vanniktech.maven.publish)

Snapshots can be found [here](https://oss.sonatype.org/#nexus-search;quick~gradle-maven-publish-plugin).

### Setting properties

To configure the coordinates of your published artifacts as well as the POM this plugin 
uses Gradle properties. It's generally recommended to set them in your `gradle.properties`
file.

> There are 3 required properties:

Reviewer: ```suggestion
There are three required properties:
```

```
GROUP=com.test.mylibrary
POM_ARTIFACT_ID=mylibrary-runtime
VERSION_NAME=3.0.5
```

> In addition to that there are some optional properties to give more details about your project:

Reviewer: ```suggestion
In addition, there are some optional properties to give more details about your project:
```

POM_SCM_DEV_CONNECTION=scm:git:ssh://git@github.com/username/mylibrary.git

POM_LICENCE_NAME=The Apache Software License, Version 2.0
POM_LICENCE_URL=https://www.apache.org/licenses/LICENSE-2.0.txt
POM_LICENCE_DIST=repo

POM_DEVELOPER_ID=username
POM_DEVELOPER_NAME=User Name
POM_DEVELOPER_URL=https://github.com/username/
```

This Gradle plugin is using itself to publish any of the updates and sets the Gradle properties in 
this [gradle.properties](gradle.properties).

In multi module projects you can set most properties in the root `gradle.properties` file and
then only set the module specific ones in the submodules. For example if you have two modules 
called `runtime` and `driver` you could only set `POM_ARTIFACT_ID` and `POM_NAME` in 
`<project-dir>/runtime/gradle.properties` and `<project-dir>/driver/gradle.properties` while sharing 
the rest by putting them into `<project-dir>/gradle.properties`.

### Where to upload to

Without any further configuration the plugin has two tasks. `uploadArchives` which will upload
to Sonatype OSS (Maven Central) by default. To publish to the local maven repository on your 
machine (`~/m2/repository`) there is `installArchives`. 

> The username and password for Sonatype OSS can be provided as Gradle properties or environment

Reviewer: Let's mention which properties need to be set here for Gradle

Author: Do you mean something other than `SONATYPE_NEXUS_USERNAME` and `SONATYPE_NEXUS_PASSWORD` which are mentioned in the next line?

Reviewer: Oh right they're both sharing the same name. Then never mind

      snapshotRepositoryUrl = "https://oss.sonatype.org/content/repositories/snapshots/"
      repositoryUsername = null // This defaults to either the SONATYPE_NEXUS_USERNAME Gradle property or the system environment variable.
      repositoryPassword = null // This defaults to either the SONATYPE_NEXUS_PASSWORD Gradle property or the system environment variable.
      signing = true // GPG signing is required by mavenCentral. If you are deploying elsewhere, you can set this to false.
    }
    
    // Modify the existing installArchives task
    installArchives {
       // Configure as above.
    }

    // Add a new target, in this case it will create a uploadArchivesInternalRepo task
    internalRepo {
       // Configure as above.
    }
  }
}
```

More information can be found in the [Documentation of the Extension](src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPluginExtension.kt)

__Note:__ To prevent looping behavior, especially in Kotlin projects / modules, you need to run the `uploadArchives` task with `--no-daemon`and `--no-parallel` flags:

`./gradlew uploadArchives --no-daemon --no-parallel`

### Signing

> The plugin supports setting up all your artifacts to be signed with GPG. Since that is a requirement

Reviewer: ```suggestion
The plugin supports signing all of your artifacts with GPG. This is a requirement when publishing to Maven Central - our default behavior. Signing parameters can be configured via:
```

> for our default publishing target Maven Central, signing is enabled by default. For that please add the following 

Reviewer: ```suggestion

```

> Gradle properties.

Reviewer: ```suggestion

```
